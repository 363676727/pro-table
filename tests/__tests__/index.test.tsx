import { render } from 'enzyme';
import React from 'react';
import { Button, Input } from 'antd';
import ProTable, { TableDropdown } from '../../src/index';
import { columns, request } from './demo';

describe('BasicTable', () => {
  it('🎏 base use', () => {
    const html = render(
      <ProTable
        size="small"
        columns={columns}
        request={request}
        rowKey="key"
        params={{ keyword: 'test' }}
        pagination={{
          defaultCurrent: 10,
        }}
        toolBarRender={action => [
          <Input.Search
            style={{
              width: 200,
            }}
          />,
          <Button
            onClick={() => {
              action.reload();
            }}
            key="1"
            type="primary"
            style={{
              marginRight: 8,
            }}
          >
            刷新
          </Button>,
          <Button
            key="2"
            style={{
              marginRight: 8,
            }}
            onClick={() => {
              action.setCurrent(3);
            }}
            type="dashed"
          >
            跳转到第三页
          </Button>,
          <Button
            key="3"
            onClick={() => {
              action.resetPageIndex();
            }}
            type="default"
            style={{
              marginRight: 8,
            }}
          >
            重置
          </Button>,
          <TableDropdown.Button
            menus={[{ key: 'copy', name: '复制' }, { key: 'clear', name: '清空' }]}
          >
            更多操作
          </TableDropdown.Button>,
        ]}
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('🎏 do not render Search ', () => {
    const html = render(
      <ProTable
        size="small"
        columns={columns}
        request={request}
        rowKey="key"
        search={false}
        params={{ keyword: 'test' }}
        pagination={{
          defaultCurrent: 10,
        }}
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('🎏  do not render default option', () => {
    const html = render(
      <ProTable
        size="small"
        options={{
          fullScreen: false,
          reload: false,
          setting: false,
        }}
        columns={[
          {
            dataIndex: 'money',
            valueType: 'money',
          },
        ]}
        request={request}
        rowKey="key"
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('🎏  do not render setting', () => {
    const html = render(
      <ProTable
        size="small"
        options={{
          fullScreen: true,
          reload: true,
          setting: false,
        }}
        columns={[
          {
            dataIndex: 'money',
            valueType: 'money',
          },
        ]}
        request={request}
        rowKey="key"
      />,
    );
    expect(html).toMatchSnapshot();
  });
});
