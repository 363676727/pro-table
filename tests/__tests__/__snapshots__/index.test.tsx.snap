// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTable üéè base use 1`] = `
<div
  class="ant-pro-table"
>
  <div
    class="ant-pro-table-toolbar"
  >
    <div
      class="ant-pro-table-toolbar-item"
    >
      <span
        class="ant-input-search ant-input-affix-wrapper"
        style="width:200px"
      >
        <input
          class="ant-input"
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <i
            aria-label="icon: search"
            class="anticon anticon-search ant-input-search-icon"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </i>
        </span>
      </span>
    </div>
    <div
      class="ant-pro-table-toolbar-item"
    >
      <button
        class="ant-btn ant-btn-primary"
        style="margin-right:8px"
        type="button"
      >
        <span>
          Âà∑ Êñ∞
        </span>
      </button>
    </div>
    <div
      class="ant-pro-table-toolbar-item"
    >
      <button
        class="ant-btn ant-btn-dashed"
        style="margin-right:8px"
        type="button"
      >
        <span>
          Ë∑≥ËΩ¨Âà∞Á¨¨‰∏âÈ°µ
        </span>
      </button>
    </div>
    <div
      class="ant-pro-table-toolbar-item"
    >
      <button
        class="ant-btn ant-btn-default"
        style="margin-right:8px"
        type="button"
      >
        <span>
          Èáç ÁΩÆ
        </span>
      </button>
    </div>
    <div
      class="ant-pro-table-toolbar-item"
    >
      <button
        class="ant-btn ant-dropdown-trigger ant-pro-table-dropdown"
        type="button"
      >
        <span>
          Êõ¥Â§öÊìç‰Ωú 
        </span>
        <i
          aria-label="icon: down"
          class="anticon anticon-down"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </i>
      </button>
    </div>
  </div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-small ant-table-empty ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                  <col
                    style="width:100px;min-width:100px"
                  />
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Â∫èÂè∑
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            ËæπÊ°ÜÂ∫èÂè∑
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Name
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            sex
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Age
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class="ant-table-row-cell-break-word"
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Address
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            money
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            date
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            dateTime
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            time
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            Áä∂ÊÄÅ
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        class="ant-table-header-column"
                      >
                        <div>
                          <span
                            class="ant-table-column-title"
                          >
                            option
                          </span>
                          <span
                            class="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                />
              </table>
            </div>
            <div
              class="ant-table-placeholder"
            >
              <div
                class="ant-empty ant-empty-normal"
              >
                <div
                  class="ant-empty-image"
                >
                  <svg
                    height="41"
                    viewBox="0 0 64 41"
                    width="64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      transform="translate(0 1)"
                    >
                      <ellipse
                        cx="32"
                        cy="33"
                        fill="#F5F5F5"
                        rx="32"
                        ry="7"
                      />
                      <g
                        fill-rule="nonzero"
                        stroke="#D9D9D9"
                      >
                        <path
                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                        />
                        <path
                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                          fill="#FAFAFA"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <p
                  class="ant-empty-description"
                >
                  No Data
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
